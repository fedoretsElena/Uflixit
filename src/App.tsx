import React from 'react';
import { BrowserRouter as Router, Redirect, Route, Switch } from 'react-router-dom';

import './App.scss';

import './core/HttpInterceptor';

import Home from './components/Home';
import Header from './components/Header';
import Footer from './components/Footer';
import Login from './components/Login/Login';
import MediaDetail from './components/MediaDetail';
import SearchResults from './components/SearchResults';

function App() {
    return (
        <div>
            <Router>
                <Header/>

                <div className="container py-4">

                    <Switch>
                        <Route exact
                               path='/home'
                               component={Home}/>

                        <Route exact
                               path='/tv-shows/:id'
                               component={MediaDetail}/>

                        <Route exact
                               path='/movies/:id'
                               component={MediaDetail}/>

                        <Route path='/search'
                               component={SearchResults}/>

                        <Route path='/login'
                               component={Login}/>

                        <Route path='/'
                               render={() => <Redirect to='/home'/> }
                        />
                    </Switch>

                </div>

                <Footer/>
            </Router>
        </div>
    );
}

export default App;

