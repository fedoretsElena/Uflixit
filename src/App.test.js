import React from "react";
import { render } from "react-testing-library";

import App from "./App";

it("App snapshot", () => {
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
});
