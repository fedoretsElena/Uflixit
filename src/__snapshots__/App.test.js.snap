// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="header bg-dark py-1"
    >
      <div
        class="container d-flex justify-content-between align-items-center"
      >
        <a
          class="header__logo"
          href="/home"
        >
          <img
            alt="logo"
            class="img-fluid"
            src="https://www.userlogos.org/files/netflix-n-logo-png.png"
          />
        </a>
        <form
          class="search-form d-flex align-items-center"
        >
          <svg
            class="search-form__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M443.5 420.2L336.7 312.4c20.9-26.2 33.5-59.4 33.5-95.5 0-84.5-68.5-153-153.1-153S64 132.5 64 217s68.5 153 153.1 153c36.6 0 70.1-12.8 96.5-34.2l106.1 107.1c3.2 3.4 7.6 5.1 11.9 5.1 4.1 0 8.2-1.5 11.3-4.5 6.6-6.3 6.8-16.7.6-23.3zm-226.4-83.1c-32.1 0-62.3-12.5-85-35.2-22.7-22.7-35.2-52.9-35.2-84.9 0-32.1 12.5-62.3 35.2-84.9 22.7-22.7 52.9-35.2 85-35.2s62.3 12.5 85 35.2c22.7 22.7 35.2 52.9 35.2 84.9 0 32.1-12.5 62.3-35.2 84.9-22.7 22.7-52.9 35.2-85 35.2z"
            />
          </svg>
          <div
            class="input"
          >
            <input
              class="input__el w-100"
              placeholder="Search... "
              type="text"
              value=""
            />
          </div>
        </form>
      </div>
    </header>
    <div
      class="container py-4"
    >
      <h4
        class="mt-5 mb-3"
      >
        Popular shows
        <span
          class="ml-2 badge badge-primary"
        >
          0
        </span>
      </h4>
      <div
        class="whirlpool-wrapper d-flex align-items-center"
      >
        <div
          class="whirlpool"
        >
          <div
            class="ring ring1"
          />
          <div
            class="ring ring2"
          />
          <div
            class="ring ring3"
          />
          <div
            class="ring ring4"
          />
          <div
            class="ring ring5"
          />
          <div
            class="ring ring6"
          />
          <div
            class="ring ring7"
          />
          <div
            class="ring ring8"
          />
          <div
            class="ring ring9"
          />
        </div>
      </div>
      <h4
        class="mt-5 mb-3"
      >
        Wanted movies
        <span
          class="ml-2 badge badge-primary"
        >
          0
        </span>
      </h4>
      <div
        class="whirlpool-wrapper d-flex align-items-center"
      >
        <div
          class="whirlpool"
        >
          <div
            class="ring ring1"
          />
          <div
            class="ring ring2"
          />
          <div
            class="ring ring3"
          />
          <div
            class="ring ring4"
          />
          <div
            class="ring ring5"
          />
          <div
            class="ring ring6"
          />
          <div
            class="ring ring7"
          />
          <div
            class="ring ring8"
          />
          <div
            class="ring ring9"
          />
        </div>
      </div>
    </div>
    <footer
      class="footer bg-dark py-4 position-absolute text-white w-100"
    >
      <div
        class="container d-flex justify-content-center"
      >
        Still need to do:
        <ul
          class="mb-0"
        >
          <li>
            Authorization
          </li>
          <li>
            <s>
              Search
            </s>
          </li>
          <li>
            Lazy loading
          </li>
        </ul>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
